From 0406d42533c3bba0b65b55a46e9d8bfe9c455ac0 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Mon, 29 Jul 2024 12:11:11 +0800
Subject: [PATCH 28/31] [AOSC] Build LoongArch64 LSX optimizations in libpng

These files were added in Chromium Change Number 5454197, but not referenced in BUILD.gn.

Update BUILD.gn to compile them when appropriate.
---
 third_party/libpng/BUILD.gn | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/third_party/libpng/BUILD.gn b/third_party/libpng/BUILD.gn
index 81648e3b2a..3116ea0e77 100644
--- a/third_party/libpng/BUILD.gn
+++ b/third_party/libpng/BUILD.gn
@@ -4,6 +4,7 @@
 
 import("//build/config/arm.gni")
 import("//build/config/mips.gni")
+import("//build/config/loongarch64.gni")
 
 if (build_with_chromium) {
   import("//testing/libfuzzer/fuzzer_test.gni")
@@ -91,6 +92,11 @@ source_set("libpng_sources") {
       "mips/filter_msa_intrinsics.c",
       "mips/mips_init.c",
     ]
+  } else if (current_cpu == "loong64" && loongarch64_use_lsx) {
+    sources += [
+      "loongarch/filter_lsx_intrinsics.c",
+      "loongarch/loongarch_lsx_init.c",
+    ]
   }
 
   configs -= [ "//build/config/compiler:chromium_code" ]
-- 
2.46.0

